Source/missiles.cpp

```diff
@@ -393,9 +393,12 @@ void CheckMissileCol(Missile &missile, int minDamage, int maxDamage, bool isDama
 	if (mid > 0 || (mid != 0 && Monsters[abs(mid) - 1].mode == MonsterMode::Petrified)) {
 		mid = abs(mid) - 1;
 		if (missile.IsTrap()

```

Comment:
This might be better written as:
```cpp
if (missile.IsTrap()
    || (missile._micaster == TARGET_PLAYERS && ( // or was fired by a monster
        Monsters[mid].isPlayerOwned() != Monsters[missile._misource].isPlayerOwned() // between monster of different alignments
        || (Monsters[mid).flags & MFLAG_BERSERK) != 0 || (Monsters[missile._misource].flags & MFLAG_BERSERK) != 0)) { // or one of them is baserked
```